plugins/inputs/t128_graphql/t128_graphql_test.go

```diff
@@ -0,0 +1,442 @@
+					  	}
+					}]
+				}
+			}
+		}`},
+		ExpectedMetrics: nil,
+		ExpectedErrors: []string{
+			"found empty data for metric test-metric: field test-field",
+		},
+	},
+	{
+		Name:       "converts tag to string if numeric",
+		EntryPoint: "allRouters[name:ComboEast]/nodes/nodes[name:east-combo]/nodes/arp/nodes",
+		Fields:     map[string]string{"test-field": "test-field"},
+		Tags:       map[string]string{"test-tag": "test-tag"},
+		Query:      ValidQuerySingleTag,
+		Endpoint: Endpoint{"/api/v1/graphql", 200, ValidExpectedRequestSingleTag, `{
+			"data": {
+				"allRouters": {
+				  	"nodes": [{
+					  	"nodes": {
+							"nodes": [{
+								"arp": {
+							  		"nodes": [{
+								  		"test-field": 128,
+								  		"test-tag": 128
+									}]
+								}
+						  	}]
+					  	}
+					}]
+				}
+			}
+		}`},
+		ExpectedMetrics: []*testutil.Metric{
+			{
+				Measurement: "test-metric",
+				Tags:        map[string]string{"test-tag": "128"},
+				Fields:      map[string]interface{}{"test-field": 128.0},
+			},
+		},
+		ExpectedErrors: nil,
+	},
+	{
+		Name:       "uses multiple tags",
+		EntryPoint: "allRouters[name:ComboEast]/nodes/nodes[name:east-combo]/nodes/arp/nodes",
+		Fields:     map[string]string{"test-field": "test-field"},
+		Tags:       map[string]string{"test-tag-1": "test-tag-1", "test-tag-2": "test-tag-2"},
+		Query:      ValidQueryDoubleTag,
+		Endpoint: Endpoint{"/api/v1/graphql", 200, ValidExpectedRequestDoubleTag, `{
+			"data": {
+				"allRouters": {
+				  	"nodes": [{
+					  	"nodes": {
+							"nodes": [{
+								"arp": {
+							  		"nodes": [{
+								  		"test-field": 128,
+										"test-tag-1": "test-string-1",
+										"test-tag-2": "test-string-2"
+									}]
+								}
+						  	}]
+					  	}
+					}]
+				}
+			}
+		}`},
+		ExpectedMetrics: []*testutil.Metric{
+			{
+				Measurement: "test-metric",
+				Tags:        map[string]string{"test-tag-1": "test-string-1", "test-tag-2": "test-string-2"},
+				Fields:      map[string]interface{}{"test-field": 128.0},
+			},
+		},
+		ExpectedErrors: nil,
+	},
+	{
+		Name:       "uses multiple tags with some none value",
+		EntryPoint: "allRouters[name:ComboEast]/nodes/nodes[name:east-combo]/nodes/arp/nodes",
+		Fields:     map[string]string{"test-field": "test-field"},
+		Tags:       map[string]string{"test-tag-1": "test-tag-1", "test-tag-2": "test-tag-2"},
+		Query:      ValidQueryDoubleTag,
+		Endpoint: Endpoint{"/api/v1/graphql", 200, ValidExpectedRequestDoubleTag, `{
+			"data": {
+				"allRouters": {
+				  	"nodes": [{
+					  	"nodes": {
+							"nodes": [{
+								"arp": {
+							  		"nodes": [{
+								  		"test-field": 128,
+										"test-tag-1": "test-string-1",
+										"test-tag-2": null
+									}]
+								}
+						  	}]
+					  	}
+					}]
+				}
+			}
+		}`},
+		ExpectedMetrics: []*testutil.Metric{
+			{
+				Measurement: "test-metric",
+				Tags:        map[string]string{"test-tag-1": "test-string-1", "test-tag-2": "test-tag-2"},
```

Comment:
this doesn't looks right - if the `test-tag-2` was null in the response, how come the metric got the value of `test-tag-2`?

Reply:
yeah this is wrong i'll fix it

Reply:
- [ ] fix test